strawberryfields/tdm/tdmprogram.py

```diff
@@ -376,30 +377,52 @@ def __init__(self, N, name=None):
             self.N = [N]
         else:
             self.N = N
-        self.concurr_modes = sum(self.N)
+        self._concurr_modes = sum(self.N)
 
-        super().__init__(num_subsystems=self.concurr_modes, name=name)
+        super().__init__(num_subsystems=self._concurr_modes, name=name)
 
-        self.type = "tdm"
+        self._type = "tdm"
         self._is_space_unrolled = False
 
-        self.timebins = 0
-        self.spatial_modes = 0
-        self.measured_modes = []
+        self._timebins = 0
+        self._spatial_modes = 0
+        self._measured_modes = set()
+
         self.rolled_circuit = None
         # `unrolled_circuit` only contains the unrolled single-shot circuit
         self.unrolled_circuit = None
         # `space_unrolled_circuit` only contains the space-unrolled single-shot circuit
         self.space_unrolled_circuit = None
         # `added_subsystems` corresponds to the number of subsystems added when space-unrolling
-        self.added_subsystems = 0
+
+        self._added_subsystems = 0
         self.run_options = {}
         """dict[str, Any]: dictionary of default run options, to be passed to the engine upon
         execution of the program. Note that if the ``run_options`` dictionary is passed
         directly to :meth:`~.Engine.run`, it takes precedence over the run options specified
         here.
         """
 
+    @property
+    def measured_modes(self):
+        """The number of measured modes in the program returned as a list."""
+        return list(self._measured_modes)
+
+    @property
+    def timebins(self):
+        """The number of timebins in the program."""
+        return self._timebins
+
+    @property
+    def spatial_modes(self):
+        """The number of spatial modes in the program."""
+        return self._spatial_modes
+
+    @property
+    def concurr_modes(self):
+        """The number of concurrent modes in the program."""
+        return self._concurr_modes
+
```

Comment:
Out of curiosity, what is the idea behind a property method that will simply return an instance variable that has the same name, only with an underscore?

Reply:
This is just Pythons way of saying that `self._concurr_modes` is private (and shouldn't be changed manually outside of the class). By having a property, the value can be retrieved but not set (you'd need to add a "setter" method as well to do that, which could involve extra validation etc.).